server/gitrest/src/routes/git/refs.ts

```diff
@@ -153,12 +169,24 @@ export function create(
         handleResponse(resultP, response);
     });
 
-    router.get("/repos/:owner/:repo/git/refs/*", (request, response, next) => {
+    router.get("/repos/:owner/:repo/git/refs/heads/:refId", (request, response, next) => {
         const resultP = getRef(
             repoManager,
             request.params.owner,
             request.params.repo,
-            getRefId(request.params[0]),
+            getRefId(request.params.refId),
+            undefined,
+            externalStorageManager);
+        handleResponse(resultP, response);
+    });
+
+    router.get("/repos/:owner/:repo/git/refs/heads/:refId/:config", (request, response, next) => {
```

Comment:
Not certain of API best practices here, but it _feels_ like a stringified JSON should be passed in the `query` params, rather than as a piece of the route.

Also, I don't think this route needs to be split into 2, regardless of config being a route param or a query param. If it's a route param, make it optional (`:config?`). Then, `getReadParams` is already handling `undefined` input, so it should all work smoothly